File: 프로그래머스/Level1/크레인 인형뽑기 게임.py
# https://programmers.co.kr/learn/courses/30/lessons/64061

def solution(board, moves):
    n = len(board)
    new_board = [([0] * n) for _ in range(n)]
    for i in range(n):
        for j in range(n):
            new_board[j][i] = board[i][j]
    result = []
    answer = 0
    for move in moves:
        move -= 1
        temp = new_board[move]
        empty = True
        for i in range(n):
            if temp[i] != 0:
                empty = False
                break
        if empty == True:
            continue
        get = temp[i]
        if len(result) != 0 and result[-1] == get:
            answer += 2
            result.pop()
        else:
            result.append(get)
        temp[i] = 0
    return answer

File: 프로그래머스/Level1/크레인 인형뽑기 게임.js
// https://programmers.co.kr/learn/courses/30/lessons/64061

const solution = (board, moves) => {
  let stack = [];
  moves
    .map((m) => m - 1)
    .forEach((m) => {
      for (let i = 0; i < board.length; i++) {
        if (board[i][m] !== 0) {
          stack.push(board[i][m]);
          board[i][m] = 0;
          break;
        }
      }
    });
  let answer = 0;
  while (true) {
    let check = false;
    for (let i = 0; i < stack.length - 1; i++) {
      if (stack[i] === stack[i + 1]) {
        stack.splice(i, 2);
        answer += 2;
        check = true;
        break;
      }
    }
    if (check) continue;
    break;
  }
  return answer;
};
